### PR TITLE
Only show non-hidden options in --help-all for options where supportl…

### LINKED ISSUE #13385: --help-all shows hidden options for nonsupported options when mapper exists
### Describe the bug

see headling

### Version

latest

### Expected behavior

options flagged with hidden() are not shown in the cli help even when their category is != SUPPORTED (e.g. Experimental options)

### Actual behavior

Hidden options are also shown

### How to Reproduce?

_No response_

### Anything else?

Will provide a PR, fixing e.g. https://github.com/keycloak/keycloak/blob/main/quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java#L53 and other storage options that should be hidden.

### PR DESCRIPTION
**Pull request by DGuhr**
…evel  is not SUPPORTED

No need to change the approval.txt files for now as the user facing options are the same but the internal logic is now right

Closes #13385

### CODE REVIEW

File: quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java [hunk 1]
```diff
@@ -321,40 +319,34 @@ public String getProvider() {
...
             .category(OptionCategory.STORAGE)
             .description("Sets the password of the Infinispan user.")
-            .hidden()
             .build();
 
     public static final Option<Boolean> STORAGE_HOTROD_CACHE_CONFIGURE = new OptionBuilder<>("storage-hotrod-cache-configure", Boolean.class)
             .category(OptionCategory.STORAGE)
             .defaultValue(true)
             .description("When set to true, Keycloak will create and configure Infinispan caches on startup.")
-            .hidden()
```
Comment (hmlnarik) [inline quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java:349]: this should be kept

File: quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java [hunk 2]
```diff
@@ -321,40 +319,34 @@ public String getProvider() {
...
             .build();
 
     public static final Option<String> STORAGE_HOTROD_CACHE_REINDEX = new OptionBuilder<>("storage-hotrod-cache-reindex", String.class)
             .category(OptionCategory.STORAGE)
             .defaultValue("all")
             .expectedValues(Stream.concat(Stream.of("all"), AutogeneratedHotRodDescriptors.ENTITY_DESCRIPTOR_MAP.values().stream().map(HotRodEntityDescriptor::getCacheName).distinct()).collect(Collectors.toList()))
             .description("List of cache names that should be indexed on Keycloak startup. Defaulting to `all` which means all caches are reindexed.")
-            .hidden()
```
Comment (hmlnarik) [inline quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java:357]: This should be kept

**Review by hmlnarik — COMMENTED** [review]
Thank you for the PR!

I agree with most of the changes; please see inline.

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testStartOptimizedHelpAll.windows.approved.txt [hunk 3]
```diff
@@ -27,13 +27,7 @@ Storage (Experimental):
...
                        Will default to a random number generated when starting the server which is
                        secure but will lead to problems when a loadbalancer without sticky sessions
-                       is used or nodes are restarted.
---storage-hotrod-cache-configure <true|false>
-                     Experimental: When set to true, Keycloak will create and configure Infinispan
-                       caches on startup. Default: true.
---storage-hotrod-cache-reindex <[cache1,cache2,...]|all>
-                     Experimental: List of cache names that should be indexed on Keycloak startup.
-                       Defaulting to `all` which means all caches are reindexed. Default: all.
+                       is used or nodes are restarted.cd quar
```
Comment (hmlnarik) [inline quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testStartOptimizedHelpAll.windows.approved.txt:30]: `cd quar`?

Comment (DGuhr) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testStartOptimizedHelpAll.windows.approved.txt:30]: oh oops... let me fix that.

Comment (DGuhr) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/approvals/cli/help/HelpCommandTest.testStartOptimizedHelpAll.windows.approved.txt]: done

**Review by hmlnarik — APPROVED** [review]

**Comment by DGuhr** [comment]
also created https://github.com/keycloak/keycloak/issues/13455 as a followup to change the implicit behaviour of "description" and add an internal flag instead.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**